src/ASM/ASMs2D.C

```diff
@@ -2046,6 +2047,9 @@ bool ASMs2D::integrate (Integrand& integrand,
         fe.iel = MLGE[iel];
         if (!this->isElementActive(fe.iel,time.t))
           continue; // zero-area or inactive element
+        if (!myElms.empty() &&
+            std::find(myElms.begin(), myElms.end(), iel) == myElms.end())
+          continue;
```

Comment:
Since this is a common operation for most (all?) of the ASM-classes, consider a method in ASMbase for this, e.g., `ASMbase::isElementInPartition(int iel) const;`

Reply:
done.